src/orng.cpp

```diff
@@ -593,8 +596,17 @@ ACTION orng::cleanup(eosio::name oracle) {
     auto oit = oracles_table.require_find(oracle.value, "unknown oracle");
     check(!oit->suspended, "oracle suspended");
     
-    undelivered_table_type undelivered_table(get_self(), get_self().value);
+    // Check if enough time has passed since last cleanup (default 1 hour = 3600 seconds)
     auto current_time = current_time_point();
+    uint64_t last_cleanup_time = get_config(last_cleanup_index, 0);
+    uint64_t cleanup_interval = get_config(cleanup_interval_index, 3600); // default 1 hour in seconds
+    
+    if (last_cleanup_time > 0) {
+        uint64_t time_since_last = current_time.sec_since_epoch() - last_cleanup_time;
+        check(time_since_last >= cleanup_interval, "cleanup can only be called once per hour");
```

Comment:
I don't know why we want to limit this? What if the table is growing faster than the cleanup_batch_size in one hour?